### ISSUE TITLE
Show errorMessage when textField is empty

### ISSUE THREAD

**Issue by ffittschen** [Skyscanner/SkyFloatingLabelTextField#23]
Hi!

Would it be possible to add a flag that allows to display the errorMessage even if the textField is empty?

In my use case I would like to display an errorMessage `A username is required to register` if the textField validation fails (because the textField is empty).

The corresponding line is [#L436 in SkyFloatingLabelTextField.swift](https://github.com/Skyscanner/SkyFloatingLabelTextField/blob/master/Sources/SkyFloatingLabelTextField.swift#L436)

It could be changed like this:
```swift
// add new property to SkyFloatingLabelTextField
/// A Bool value whether to display the error message even when the text is empty.
public var showErrorMessageOnEmptyText: Bool = false

// modify L436 to this
let showErrorMessage = self.showErrorMessageOnEmptyText && self.hasErrorMessage
let titleVisible = (titleText != nil) && (self.hasText() || showErrorMessage)
```

I could create a PR if you approve the idea.

**Comment by gergelyorosz** [comment]
Thanks for sharing your use case. I see this as something valid that the control should either support out of the box, or allow some simple extensiblity (e.g. subclassing) to achieve.

I'm thinking of whether introducing the `showErrorMessageOnEmptyText` would still keep the API generic enough. Another option would be to allow overriding e.g. the `titleVisible` property, where a subclass can change it as they see fit.

Another issue that came up - as I was testing your proposed solution - is that currently the textfield clears the error message when it gets focus (== becomes the first responder). This also feels wrong to do for this specific case - you should be able to create a textfield with an error message, and show this message from the start, when it's first displayed.

I'll do some more playing with the code to see if there's a simple way we could support this.

**Comment by ffittschen** [comment]
I just discovered that the positioning of the errorMessage would also be wrong, because the rect of the `titleLabel.frame` [gets calculated](https://github.com/Skyscanner/SkyFloatingLabelTextField/blob/master/Sources/SkyFloatingLabelTextField.swift#L563) using `self.hasText()` instead of `self.editingOrSelected`

**Comment by gergelyorosz** [comment]
Very true! Good point - I'll look into fixing both at the same time. I'm still a bit unsure about donig it with a property or subclassing - my gut tells me you _shouldn't_ have to subclass to be able to support this use case.

I'll create a patch for this and mention you as well, so you can take a look as well.

**Comment by gergelyorosz** [comment]
It _did_ feel wrong to me to add a new property to support this edge cases.

In the end I reviewed how the error message clearing works - and what felt really wrong is why there's some "magic" clearing in the first place. If a developer sets the `errorMessage`, I would expect that to be visible until it is cleared.

So I ended up solving this issue you've experienced by removing this "automatic" resetting of `errorMessage`, as well as addressing the things you've mentioned (like the frame not calculating). [This is the PR](https://github.com/Skyscanner/SkyFloatingLabelTextField/pull/28) for th change, and hopefully it will be merged soon, with a new pod version coming out.

**Comment by intonarumori** [comment]
In the first implementation we had a default behaviour for text clearing:
- you can set the error message
- once the text is changed or cleared, the error get's cleared too
To create a custom behaviour we had `clearsOnTextChange` property. If set to `false` the error wouldn't be cleared automatically and you could handle it yourself.

I liked this setup since most of the time this is what you want and you don't have to implement it for every single text input, it's easier for newcomers. For more complex behaviour you could always turn off the `clearsOnTextChange` flag.

Let me know what you think.

**Comment by gergelyorosz** [comment]
This issue should be fixed in v1.2.0. Please let me know if you still see any issues, and I can reopen this. Closing this one for now.
